File: src/my_py_pkg/my_py_pkg/Hardware_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus

class HwstatuspublisherNode(Node): 
    def __init__(self):
        super().__init__("hw_status") 
        self.hardware_status_publisher = self.create_publisher(HardwareStatus,"hw_status",10)
        self.timer = self.create_timer(1.0,self.publish_status)
        self.get_logger().info(" publsiher is publishing :)")

    def publish_status(self):
        msg = HardwareStatus()
        msg.temperature = 49
        msg.are_motors_ready = True
        msg.debug_message = "Nothing special here"
        self.hardware_status_publisher.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = HwstatuspublisherNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/Hardware_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HwstatuspublsihNode : public rclcpp::Node 
{
public:
    HwstatuspublsihNode() : Node("hw_status") 
    {
        publisher = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>
                       ("hw_status",10);
        timer = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&HwstatuspublsihNode::PublishNews,this));
        RCLCPP_INFO(this ->get_logger(),"Transmission started");
    }

private:
void PublishNews(){
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.temperature = 50;
        msg.are_motors_ready = true;
        msg.debug_message = "Nothing here :)";
        publisher->publish(msg);
    }
      rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher;
      rclcpp::TimerBase::SharedPtr timer;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HwstatuspublsihNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}